quran/fields.py

```diff
@@ -0,0 +1,82 @@
+from django.core.exceptions import ValidationError
+from django.db import models
+
+
+class SegmentsField(models.TextField):
+    description = "Segments"
+
+    def to_python(self, value):
+        if value is None:
+            return None
+        # if value is empty string return empty list
+        if value == '':
+            return []
+        # if value is list
+        if isinstance(value, list):
+            # if list is not empty
+            if value:
+                # validate list of tuples
+                self._validate_tuples_containing_in_list(value)
+                self._validate_length_of_segments(
+                    value, "Length of tuples must be equals 2")
+            return value
+        try:
+            # try to parse a string of segments from DB to a list of tuples
+            # "12:15,16:19" => [(12, 15), (16, 19)]
+            segments_list = [tuple(
+                int(timecode) for timecode in timecodes.split(':'))
+                for timecodes in value.split(',')]
+            # validate list of tuples
+            self._validate_length_of_segments(
+                segments_list, 'Invalid string of segments')
+            return segments_list
+
+        except (TypeError, ValueError, IndexError):
+            raise ValidationError(
+                message='Invalid string of segments',
+                code='invalid')
+
+    def from_db_value(self, value, expression, connection):
+        return self.to_python(value)
+
+    def get_prep_value(self, value):
+        # save to DB None or empty string
```

Comment:
Add this magical line just in case:
```py
value = super().get_prep_value(value)
```
In general, this function can be simplified:
```py
value = super().get_prep_value(value)
value = self.to_python(value)  # <- this function has all checks incapsulated
return ','.join(f'{segment[0]}:{segment[1]}' for segment in value)
```